src/Integration.UnitTests/LocalServices/RuleSetInspectorTests.cs

```diff
@@ -0,0 +1,482 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RuleSetInspectorTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.LocalServices
+{
+    [TestClass]
+    public class RuleSetInspectorTests
+    {
+        /* Notes: "By default" is referred to the way we create rulesets for projects when binding.
+        Also, please read the comments inside FindConflictingRules to better understand the merge logic (if needed to)
+        */
+        private const int DefaultNumberOfRules = 3;
+
+        private RuleSetInspector testSubject;
+        private ConfigurableServiceProvider serviceProvider;
+        private ConfigurableVsShell shell;
+        private ConfigurableVsGeneralOutputWindowPane outputPane;
+
+        #region Test plumbing
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.serviceProvider = new ConfigurableServiceProvider();
+
+            this.outputPane = new ConfigurableVsGeneralOutputWindowPane();
+            this.serviceProvider.RegisterService(typeof(SVsGeneralOutputWindowPane), this.outputPane);
+
+            this.shell = new ConfigurableVsShell();
+            this.shell.RegisterPropertyGetter((int)__VSSPROPID2.VSSPROPID_InstallRootDir, () => this.SonarQubeRuleSetFolder);
+            this.serviceProvider.RegisterService(typeof(SVsShell), this.shell);
+           
+            this.testSubject = new RuleSetInspector(this.serviceProvider);
+
+            if (!Directory.Exists(VsRuleSetsDirectory))
```

Comment:
Missing _this_.  _CreateDirectory_ does an _exists_ check internally and no-ops it the directory already exists.